src/lib/storage/encoding_type.hpp

```diff
@@ -67,4 +68,21 @@ struct SegmentEncodingSpec {
 
 using ChunkEncodingSpec = std::vector<SegmentEncodingSpec>;
 
+constexpr std::array<EncodingType, 6> all_encoding_types = {
```

Comment:
Shouldn't the size be deducible?

Reply:
Is there is nicer way to help the compiler deduce the type than that?
```
constexpr std::array all_segment_encoding_specs{std::initializer_list<SegmentEncodingSpec>{
    {EncodingType::Unencoded},
    {EncodingType::Dictionary, VectorCompressionType::FixedSizeByteAligned},
    {EncodingType::Dictionary, VectorCompressionType::SimdBp128},
    {EncodingType::FrameOfReference, VectorCompressionType::FixedSizeByteAligned},
    {EncodingType::FrameOfReference, VectorCompressionType::SimdBp128},
    {EncodingType::FixedStringDictionary, VectorCompressionType::FixedSizeByteAligned},
    {EncodingType::FixedStringDictionary, VectorCompressionType::SimdBp128},
    {EncodingType::LZ4, VectorCompressionType::FixedSizeByteAligned},
    {EncodingType::LZ4, VectorCompressionType::SimdBp128},
    {EncodingType::RunLength}}};
```

Reply:
Uh, not sure. Works for me.